v2/rest/client.go

```diff
@@ -0,0 +1,310 @@
+}
+
+func (c *client) CreateNode(ctx context.Context, node models.CreateNodeRequest) (models.CreateNodeResponse, error) {
+	request := rest.Post("v2/nodes").
+		WithJSONPayload(node).
+		SetHeader("Accept", "application/json").
+		SetHeader("X-Client-Id", c.clientID)
+
+	var response models.CreateNodeResponse
+	if err := c.DoAndUnmarshal(ctx, request, &response); err != nil {
+		return models.CreateNodeResponse{}, err
+	}
+
+	return response, nil
+}
+
+func (c *client) UpdateNode(ctx context.Context, id uuid.UUID, node models.UpdateNodeRequest) (models.UpdateNodeResponse, error) {
+	request := rest.Patch("nodes/{node}").
+		Assign("node", id).
+		WithJSONPayload(node).
+		SetHeader("Accept", "application/json").
+		SetHeader("X-Client-Id", c.clientID)
+
+	var response models.UpdateNodeResponse
+	if err := c.DoAndUnmarshal(ctx, request, &response); err != nil {
+		return models.UpdateNodeResponse{}, err
+	}
+
+	return response, nil
+}
+
+func (c *client) DeleteNode(ctx context.Context, id uuid.UUID) (err error) {
+	request := rest.Delete("v2/nodes/{node}").
+		Assign("node", id).
+		SetHeader("Accept", "application/json").
+		SetHeader("X-Client-Id", c.clientID)
+
+	_, err = c.Do(ctx, request)
+
+	return
+}
+
+func (c *client) DuplicateNode(ctx context.Context, source uuid.UUID, destination uuid.UUID, suffix string) (models.DuplicateNodeResponse, error) {
+	request := rest.Post("v2/nodes/{node}/duplicate{?dstParentNodeId,label_suffix}").
+		Assign("node", source).
+		Assign("dstParentNodeId", destination).
+		Assign("label_suffix", suffix).
+		SetHeader("Accept", "application/json").
+		SetHeader("X-Client-Id", c.clientID)
+
+	var response models.DuplicateNodeResponse
+	if err := c.DoAndUnmarshal(ctx, request, &response); err != nil {
+		return models.DuplicateNodeResponse{}, err
+	}
+
+	return response, nil
+}
+
+func (c *client) GetAncestors(ctx context.Context, id uuid.UUID, height int, nodeTypes ...string) (models.GetAncestorsResponse, error) {
+	request := rest.Get("v2/nodes/{node}/ancestors{?height,type*}").
+		Assign("node", id).
+		Assign("height", height).
+		Assign("type", nodeTypes).
+		SetHeader("Accept", "application/json").
+		SetHeader("X-Client-Id", c.clientID)
+
+	var response models.GetAncestorsResponse
+	if err := c.DoAndUnmarshal(ctx, request, &response); err != nil {
+		return models.GetAncestorsResponse{}, err
+	}
+
+	return response, nil
+}
+
+func (c *client) GetCompany(ctx context.Context, id uuid.UUID) (models.GetCompanyResponse, error) {
+	request := rest.Get("v2/nodes/{node}/company").
+		Assign("node", id).
+		SetHeader("Accept", "application/json").
+		SetHeader("X-Client-Id", c.clientID)
+
+	var response models.GetCompanyResponse
+	if err := c.DoAndUnmarshal(ctx, request, &response); err != nil {
+		return models.GetCompanyResponse{}, err
+	}
+
+	return response, nil
+}
+
+func (c *client) GetSubtree(ctx context.Context, id uuid.UUID, filter TreeFilter, continuationToken string) (models.GetSubtreeResponse, error) {
```

Comment:
Should this client handle pagination for the user? Since v2 enforces pagination I feel this can be nice, to avoid this code existing on each call site.

Reply:
chaching the pagenation token in a map? e.g. tokenchache['GetSubtree']

Reply:
I was thinking more like the GetSubtree always returns all results (by looping over the endpoint and returning until the next URL is empty)

Reply:
Yeah, that might be cool